modules/gateway/peers.go

```diff
@@ -348,6 +408,32 @@ func acceptConnVersionHandshake(conn net.Conn, version string) (remoteVersion st
 	return remoteVersion, nil
 }
 
+// acceptConnSessionHandshake performs the session handshake and should be
+// called on the side accepting a connection request.
+// if err == nil, wantConn indicates if a connection is wanted on both sides.
+func acceptConnSessionHandshake(conn net.Conn, uniqueID gatewayID, wantConn bool) (err error) {
```

Comment:
not a big deal, but we typically avoid named return values unless the meaning of the value is not obvious. Even then, we don't like naked `return` statements.